dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/kafka/metrics/MetricsTest.java

```diff
@@ -0,0 +1,153 @@
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        PropertiesAsset appConfig = new PropertiesAsset()
+                        .addProperty(AbstractKafkaTestServlet.KAFKA_BOOTSTRAP_PROPERTY, KafkaTests.kafkaContainer.getBootstrapServers())
+                        .include(ConnectorProperties.simpleOutgoingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           MetricsTestServlet.EMITTER_OUTGOING_CHANNEL, MetricsTestServlet.EMITTER_TOPIC))
+                        .include(ConnectorProperties.simpleIncomingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           MetricsReceptionBean.CHANNEL_IN, APP_NAME, MetricsTestServlet.EMITTER_TOPIC))
+                        .include(ConnectorProperties.simpleOutgoingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           MetricsDeliveryBean.METRICS_OUTGOING, MetricsTestServlet.OUTGOING_TOPIC))
+                        .include(ConnectorProperties.simpleIncomingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           MetricsReceptionBean.CHANNEL_MESSAGE_IN, APP_NAME, MetricsTestServlet.OUTGOING_TOPIC));
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addAsLibraries(kafkaClientLibs())
+                        .addAsManifestResource(kafkaPermissions(), "permissions.xml")
+                        .addPackage(MetricsTestServlet.class.getPackage())
+                        .addPackage(KafkaTestConstants.class.getPackage())
+                        .addPackage(AbstractKafkaTestServlet.class.getPackage())
+                        .addAsResource(appConfig, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @Test
+    public void testDeliverPayload() throws Exception {
+        // Test that metrics are updated when a payload is delivered via an emitter
+        runTest(server, APP_NAME + "/MetricsTest", "emitterDeliverPayload");
+        HashMap<String, Integer> metrics = getMetrics();
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-incoming"), equalTo(5));
+//      Assertion disabled as emitters currently don't report metrics per spec
+//        assertTrue("Outgoing and incoming channel counts are not the same",
+//                   metrics.get("mp_messaging_message_count_total{channel=\"MetricsEmitter\",mp_scope=\"base\",}") == metrics
+//                                   .get("mp_messaging_message_count_total{channel=\"metrics-incoming\",mp_scope=\"base\",}"));
```

Comment:
Is there an issue open for this?

Reply:
https://github.com/OpenLiberty/open-liberty/issues/26665